crc32.c

```diff
@@ -64,9 +64,15 @@ ZLIB_INTERNAL uint32_t crc32_generic(uint32_t crc, const unsigned char *buf, uin
     return crc ^ 0xffffffff;
 }
 
+#ifdef ZLIB_COMPAT
+unsigned long ZEXPORT crc32(unsigned long crc, const unsigned char *buf, unsigned int len) {
+    return (unsigned long) crc32_z((uint32_t) crc, buf, len);
+}
+#else
 uint32_t ZEXPORT PREFIX(crc32)(uint32_t crc, const unsigned char *buf, uint32_t len) {
```

Comment:
Since this branch is zlib-ng native, we should perhaps not use the PREFIX(). Either that or we should use it in both branches.
Come to think of it, we might want to use it after all to enable custom function prefixes, since some (few) users do require that to avoid clashes between a statically compiled-in zlib(-ng) and a dl()'ed zlib(-ng) from for example a 3rd party plugin. Lets keep using PREFIX() to ease the use in those cases and reduce the amount of unexpected compile/runtime failures for those users.

Reply:
Updated

Reply:
You forgot the return call to crc32_z :)

Reply:
Updated again ;)

Reply:
Doesn't the len parameter also need a cast to avoid warnings?

Reply:
"unsigned int" is either 16-bit or 32-bit, "uint32_t" is always 32-bit type. Because it would convert to same or larger size type, it doesn't need cast.

Reply:
Doh, of course it is. I was somehow reading both parameters as unsigned long. I so prefer the stdint names. :)